pyvista/examples/downloads.py

```diff
@@ -3594,7 +3770,7 @@ def download_cavity(load=True):
 
     """
     directory, _ = _download_file('OpenFOAM.zip')
-    filename = os.path.join(directory, 'cavity', 'case.foam')
+    filename = os.path.join(directory, 'OpenFOAM', 'cavity', 'case.foam')
```

Comment:
`directory` in the line above should be returning the path to the folder.  The `cavity/case.foam` are part of the internal structure of this zip file.  The resulting path I get with `examples.download_cavity(load=False)` is `'/path/to/pyvista/examples/OpenFOAM/OpenFOAM/cavity/case.foa
m'`, which shows that this top-level folder is duplicated.

Reply:
This zip file, and it seems most of the others too, include a root level folder, presumably because of the very issue motivated by this PR.  One thought is to allow an optional parameter to put the base zip name in the path, or not, but it requires exposing that parameter in many functions.  Or we can leave as is, Im fine with that too.

Reply:
> This zip file, and it seems most of the others too, include a root level folder, presumably because of the very issue motivated by this PR. One thought is to allow an optional parameter to put the base zip name in the path, or not, but it requires exposing that parameter in many functions. Or we can leave as is, Im fine with that too.

Thats the intended consequence of this PR, and I’d prefer keeping the default behavior the only behavior.  We can choose to restructure the archive files (ideally with a new file), but unzipping the contents into a single directory might overwrite identically named files in the edge case that two archives contain identically names files.